### PR TITLE
ENH simplify the tree narration removing entropy explanation

### PR DESCRIPTION
**Pull request by glemaitre**
Removing the narration linked to the internal split mechanism.
I added a couple of insights regarding the prediction.

### CODE REVIEW

**Comment by lesteve** [comment]
LGTM overall. The things I am wondering about:
- do we want to mention that the criterion can be either Gini or entropy or even skip this (I guess gini is mention in the output of `plot_tree` so maybe you need to mention it somehow)?
- not sure what to do with https://275-246063957-gh.circle-artifacts.com/0/jupyter-book/python_scripts/trees.html#the-regression-criterion. Maybe showing a tree with two splits, 4 splits, etc ... similarly to what is done for classification to build up the intuition that the tree is piece-wise constant and that the split decides where the "jumps" happen. Right now the first plot for regression does not convey that very well:
image
(i.e. the piecewise constantness does not jump at you).
- there is a "information gain" remaining somewhere, probably this can be removed and we can stay vague with a formulation like "best split according to some criterion that can be set"

File: python_scripts/trees.py [hunk 1]
```diff
@@ -185,361 +185,126 @@ def plot_decision_function(X, y, clf, ax=None):
...
+# level of the tree will be using the feature "Culmen Depth".
 
 # %%
-from sklearn.tree import plot_tree
-
-tree = DecisionTreeClassifier(criterion="entropy", max_depth=1)
-tree.fit(single_feature.to_frame(), y_train)
-_ = plot_tree(tree)
-
+tree.set_params(max_depth=2)
```
Comment (lesteve) [inline python_scripts/trees.py:283]:
```suggestion
tree = DecisionTreeClassifier(max_depth=2)
```

seems less magical this way.

**Comment by lesteve** [comment]
> (I guess gini is mention in the output of plot_tree so maybe you need to mention it somehow)?

Or maybe use `plot_tree(tree, impurity=False)` ...

**Comment by glemaitre** [comment]
> do we want to mention that the criterion can be either Gini or entropy or even skip this (I guess gini is mention in the output of plot_tree so maybe you need to mention it somehow)?

Yes we can simplify. I would just mention that we use the Gini just to point to the tree plot and that's it.

> not sure what to do with https://275-246063957-gh.circle-artifacts.com/0/jupyter-book/python_scripts/trees.html#the-regression-criterion. Maybe showing a tree with two splits, 4 splits, etc ... similarly to what is done for classification to build up the intuition that the tree is piece-wise constant and that the split decides where the "jumps" happen. Right now the first plot for regression does not convey that very well:

Yes I forgot that it was structured like this. Basically, we want to explain how it works in regression. Likewise the classification, we need 2 plots to explain the principle of predicting the mean of each partition which is in line with your comment. I will make this message to the point and reduce a bit the narration.

**Comment by glemaitre** [comment]
OK this should be better now. I apply your suggestion @lesteve

**Comment by lesteve** [comment]
Merging, thanks!

**Pull request merged**